### ISSUE TITLE
AltrootFS panics when file does not exist yet

### ISSUE THREAD

**Issue by sffc** [manuel-woelker/rust-vfs#8]
Minimal repro:

```rust
let fs = vfs::altroot::AltrootFS::new("/tmp");
fs.path("dummy");
```

If `/tmp/dummy` does not exist on your filesystem, the above code panics here:

https://github.com/manuel-woelker/rust-vfs/blob/aa05dfda7641e83b9069a3d308b863761a412a3a/src/altroot.rs#L57

It seems that [canonicalize](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.canonicalize) requires the path to exist, so if the file does not exist yet, then we panic.

I assume this is not the expected behavior, because I think calling `.path` is the one and only way to create a new file, but right now, AltrootFS panics before I can create the file.

**Comment by manuel-woelker** [comment]
Thanks for the bug report. Looks like this [1] is the root issue, AFAIC there is no clear alternative as of yet.

I am currently in the process of refactoring the API/traits [2], and I will keep this issue in mind when getting to the Altroot/Chroot implementation, and try to find a working solution.

[1] https://github.com/rust-lang/rust/issues/59117
[2] https://github.com/manuel-woelker/rust-vfs/tree/fb-erased-fs-trait

**Comment by manuel-woelker** [comment]
@sffc I just released version 0.3.0, which has completely broken the API, but should fix the issue with path canonicalization.

If you want to take it for a spin and try it out, please let me know if this fixes the problem. Thanks!

**Comment by manuel-woelker** [comment]
Closing since this should be fixed now, please feel free to open a new issue if you have any further problems.
